org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/TraceRegionMerger.java

```diff
@@ -177,7 +179,7 @@ private AbstractTraceRegion replaceTruncated(int index, AbstractTraceRegion regi
 	}
 
 	private AbstractTraceRegion replaceMerged(int index, AbstractTraceRegion region, List<ILocationData> locationData) {
-		List<ILocationData> mergedLocations = newArrayList(region.getAssociatedLocations());
+		Set<ILocationData> mergedLocations = newLinkedHashSet(region.getAssociatedLocations());
 		mergedLocations.addAll(locationData);
 		AbstractTraceRegion merged = new TraceRegion(
 				region.getMyOffset(), region.getMyLength(),
```

Comment:
Could you turn it into a list here already? Then we don't need to change the signature of TraceRegion.